#!/usr/bin/env python
PACKAGE = "ftag2"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("sobelThreshHigh", int_t, 0, "sobelThreshHigh", 100, 0, 255)
gen.add("sobelThreshLow", int_t, 0, "sobelThreshLow", 30, 0, 255)

blur_width_enum = gen.enum([
  gen.const("3", int_t, 3, "3"),
  gen.const("5", int_t, 5, "5")],
  "sobelBlurWidth")
gen.add("sobelBlurWidth", int_t, 0, "sobelBlurWidth", 3, edit_method=blur_width_enum)
  
gen.add("houghRhoRes", double_t, 0, "houghRhoRes", 1.0, 0.01, 10.0)
gen.add("houghThetaRes", double_t, 0, "houghThetaRes", 1.0, 0.01, 30.0) # *degree

gen.add("houghEdgelThetaMargin", double_t, 0, "houghEdgelThetaMargin", 15.0, 0.01, 360.0) # * degree

gen.add("houghBlurRhoWidth", double_t, 0, "houghBlurRhoWidth", 5.0, 0.0, 100.0) # * houghRhoRes
gen.add("houghBlurThetaWidth", double_t, 0, "houghBlurThetaWidth", 10.0, 0.0, 360.0) # * houghThetaRes

gen.add("houghNMSRhoWidth", double_t, 0, "houghNMSRhoWidth", 9.0, 3.0, 99.0) # * houghRhoRes
gen.add("houghNMSThetaWidth", double_t, 0, "houghNMSThetaWidth", 13.0, 3.0, 360.0) # * houghThetaRes

gen.add("houghMinAccumValue", double_t, 0, "houghMinAccumValue", 1.0, 0.0, 1000.0)
gen.add("houghMaxDistToLine", double_t, 0, "houghMaxDistToLine", 10.0, 0.01, 100.0)
gen.add("houghMinSegmentLength", double_t, 0, "houghMinSegmentLength", 0.0, 0.0, 100.0)
gen.add("houghMaxSegmentGap", double_t, 0, "houghMaxSegmentGap", 0.0, 0.0, 100.0)

gen.add("imRotateDeg", double_t, 0, "imRotateDeg", 0.0, 0.0, 360.0)

exit(gen.generate(PACKAGE, "ftag2", "CamTestbench"))
